tests/test_routes.py

```diff
@@ -0,0 +1,247 @@
+        "distance_from_the_sun": 0.39
+    }
+
+
+def test_get_one_planet_missing_record(client, two_saved_planets):
+    response = client.get("/planets/100")
+    response_body = response.get_json()
+
+    assert response.status_code == 404
+    assert response_body == {"message": "Planet 100 not found"}
+
+
+def test_get_one_planet_invalid_id(client, two_saved_planets):
+    response = client.get("planets/faraway")
+    response_body = response.get_json()
+
+    assert response.status_code == 400
+    assert response_body == {"message": "Planet faraway invalid"}
+
+
+def test_create_one_planet(client):
+    test_data = {
+        "name": "Earth",
+        "description": "A beautiful blue-green planet with life-supporting atmosphere",
+        "distance_from_the_sun": 1.00
+    }
+    response = client.post("/planets", json=test_data)
+    response_body = response.get_data(as_text=True)
+
+    assert response.status_code == 201
+    assert response_body == "Planet Earth successfully created"
+
+
+def test_create_one_planet_no_name(client):
+    test_data = {
+        "description": "A beautiful blue-green planet with life-supporting atmosphere",
+        "distance_from_the_sun": 1.00
+    }
+
+    with pytest.raises(KeyError, match="name"):
+        response = client.post("/planets", json=test_data)
+
+
+def test_create_one_planet_no_description(client):
+    test_data = {
+        "name": "Earth",
+        "distance_from_the_sun": 1.00
+    }
+
+    with pytest.raises(KeyError, match="description"):
+        response = client.post("/planets", json=test_data)
+
+
+def test_create_one_planet_no_distance(client):
+    test_data = {
+        "name": "Earth",
+        "description": "A beautiful blue-green planet with life-supporting atmosphere",
+    }
+
+    with pytest.raises(KeyError, match="distance_from_the_sun"):
+        response = client.post("/planets", json=test_data)
+
+
+def test_create_one_planet_with_extra_keys(client, two_saved_planets):
+    test_data = {
+        "surface_area": 196.9,
+        "name": "Earth",
+        "description": "A beautiful blue-green planet with life-supporting atmosphere",
+        "distance_from_the_sun": 1.00,
+        "age": 4.543,
+    }
+
+    response = client.post("/planets", json=test_data)
+    response_body = response.get_data(as_text=True)
+
+    assert response.status_code == 201
+    assert response_body == "Planet Earth successfully created"
+
+
+def test_update_planet(client, two_saved_planets):
+    test_data = {
+        "name": "Uranus",
+        "description": "A blue-green gas giant which also has a ring system",
+        "distance_from_the_sun": 19.19
+    }
+
+    response = client.put("/planets/1", json=test_data)
+    response_body = response.get_data(as_text=True)
+
+    assert response.status_code == 200
+    assert response_body == "Planet #1 successfully updated"
+
+
+def test_update_planet_with_extra_keys(client, two_saved_planets):
```

Comment:
We'd also like to test update with missing keys (as with create), and ensure that we get back 400 Bad Requests.